src/proto/h1/encode.rs

```diff
@@ -202,48 +202,51 @@ impl Encoder {
     }
 }
 
+impl<B> EncodedBuf<B>
+where
+    B: Buf,
+{
+    fn buf(&self) -> &dyn Buf {
```

Comment:
Do we know if this negatively affects performance at all?

Reply:
I can't think of a reason why it should, except that i don't have `#inline` sprinkled on the methods. I could add them, but since i don't what effect it will have, it would feel a bit cargo culty...

Reply:
totally open to input here- also I can just scrap it if there's any performance concerns

Reply:
i've put the `#[inline]`s back. the situation should now be no better or worse than before.

Reply:
no problemo